File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        set<int>st;
        unordered_map<int,int>mp;

        for(int i=0; i<n; ++i){
            mp[nums[i]] = i+1;
        }

        for(int i=0; i<n; ++i){
           int f = target - nums[i];

           if(mp.count(f)){
            int j = mp[f]-1;
            if(j!=i) return {i, j};
           }
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}
        for i in range(0, len(nums)):
            if target - nums[i] in dict:
                return [i, dict.get(target-nums[i])]
            else:
                dict[nums[i]] = i
        
        return []